foreshadow/steps/preprocessor.py

```diff
@@ -88,6 +113,17 @@ def fit_transform(self, X, *args, **kwargs):
         """
         return self.fit(X, *args, **kwargs).transform(X)
 
+    def _get_intent(self, column):
+        override_key = "_".join([Override.INTENT, column])
+        if (
+            self.cache_manager.has_override()
+            and override_key in self.cache_manager[AcceptedKey.OVERRIDE]
+        ):
+            intent = self.cache_manager[AcceptedKey.OVERRIDE][override_key]
+        else:
+            intent = self.cache_manager[AcceptedKey.INTENT][column]
+        return intent
+
     def _prepare_transformation_pipeline(self, intent, column):
         # TODO this must be some optimization we can do. Walk through both
         #  intent override cases again to check how it affects the logic.
```

Comment:
These two functions look so similar...

Reply:
I'll remove the second one. It was not used anymore.